zeebe/gateway-protocol/src/main/proto/v2/cluster-variables.yaml

```diff
@@ -0,0 +1,314 @@
+        "400":
+          $ref: 'common-responses.yaml#/components/responses/InvalidData'
+        "401":
+          $ref: 'common-responses.yaml#/components/responses/Unauthorized'
+        "403":
+          $ref: 'common-responses.yaml#/components/responses/Forbidden'
+        "404":
+          description: Cluster variable not found
+        "500":
+          $ref: 'common-responses.yaml#/components/responses/InternalServerError'
+
+components:
+  schemas:
+    ClusterVariableScopeEnum:
+      type: string
+      enum: [ GLOBAL, TENANT ]
+      description: The scope of a cluster variable - either GLOBAL or TENANT
+    CreateClusterVariableRequest:
+      type: object
+      required:
+        - name
+        - value
+      properties:
+        name:
+          type: string
+          description: The name of the cluster variable
+        value:
+          type: object
+          description: The value of the cluster variable
+    ClusterVariableResult:
+      type: object
+      properties:
+        name:
+          type: string
+        value:
+          type: string
+        scope:
+          $ref: '#/components/schemas/ClusterVariableScopeEnum'
+        tenantId:
+          type: string
+          nullable: true
+          description: Required if scope is TENANT, must be null or undefined for GLOBAL
+    ClusterVariableSearchQueryRequest:
+      description: Cluster variable search query request.
+      additionalProperties: false
+      type: object
+      allOf:
+        - $ref: 'search-models.yaml#/components/schemas/SearchQueryRequest'
+      properties:
+        sort:
+          description: Sort field criteria.
+          type: array
+          items:
+            $ref: '#/components/schemas/ClusterVariableSearchQuerySortRequest'
+        filter:
+          description: The cluster variable search filters.
+          allOf:
+            - $ref: '#/components/schemas/ClusterVariableSearchQueryFilterRequest'
+    ClusterVariableSearchQuerySortRequest:
+      type: object
+      properties:
+        field:
+          description: The field to sort by.
+          type: string
+          enum:
+            - name
+            - value
+            - tenantId
+            - scope
+        order:
+          $ref: 'search-models.yaml#/components/schemas/SortOrderEnum'
+      required:
+        - field
+    ClusterVariableSearchQueryFilterRequest:
+      description: Cluster variable filter request.
+      type: object
+      properties:
+        name:
+          description: Name of the cluster variable.
+          allOf:
+            - $ref: 'filters.yaml#/components/schemas/StringFilterProperty'
+        value:
+          description: The value of the cluster variable.
+          allOf:
+            - $ref: 'filters.yaml#/components/schemas/StringFilterProperty'
+        scope:
+          description: The scope filter for cluster variables.
+          allOf:
+            - $ref: '#/components/schemas/ClusterVariableScopeEnum'
```

Comment:
🔧 We should create a `ClusterVariableScopeEnumFilterProperty` that allows advanced search for the enum. You can use `JobKindEnum` as an example (in OpenAPI spec and REST gateway code). It needs a filter property, advanced search schema, custom deserializer, and mappings in the code to search operations. The `JobKindEnum` and `JobKindFilterProperty` will show you the way 🙂